auth/postgres/groups.go

```diff
@@ -13,471 +13,494 @@ import (
-		"limit":    limit,
-		"offset":   offset,
-		"metadata": m,
+	params, err := gr.toDBMemberPage("", "", "", pm)
+	if err != nil {
+		return auth.GroupPage{}, err
 	}
 
 	rows, err := gr.db.NamedQueryContext(ctx, q, params)
 	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
+		return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveMembership, err)
 	}
 	defer rows.Close()
 
-	var items []groups.Group
+	var items []auth.Group
 	for rows.Next() {
 		dbgr := dbGroup{}
 		if err := rows.StructScan(&dbgr); err != nil {
-			return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
+			return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveMembership, err)
 		}
-		gr, err := toGroup(dbgr)
+		gr, err := gr.toGroup(dbgr)
 		if err != nil {
-			return groups.GroupPage{}, err
+			return auth.GroupPage{}, err
 		}
 		items = append(items, gr)
 	}
 
-	cq := fmt.Sprintf(`SELECT COUNT(*) FROM thing_group_relations gr, groups g
-					   WHERE gr.group_id = g.id and gr.member_id = :userID %s;`, mq)
+	cq := fmt.Sprintf(`SELECT COUNT(*) FROM group_relations gr, groups g
+					   WHERE gr.group_id = g.id and gr.member_id = :member_id %s `, mq)
 
 	total, err := total(ctx, gr.db, cq, params)
 	if err != nil {
-		return groups.GroupPage{}, errors.Wrap(errSelectDb, err)
+		return auth.GroupPage{}, errors.Wrap(auth.ErrFailedToRetrieveMembership, err)
 	}
 
-	page := groups.GroupPage{
+	page := auth.GroupPage{
 		Groups: items,
-		PageMetadata: groups.PageMetadata{
+		PageMetadata: auth.PageMetadata{
 			Total:  total,
-			Offset: offset,
-			Limit:  limit,
+			Offset: pm.Offset,
+			Limit:  pm.Limit,
+			Size:   uint64(len(items)),
 		},
 	}
 
 	return page, nil
 }
 
-func (gr groupRepository) Assign(ctx context.Context, memberID, groupID string) error {
-	dbr, err := toDBGroupRelation(memberID, groupID)
+func (gr groupRepository) Assign(ctx context.Context, groupID, groupType string, ids ...string) error {
+	tx, err := gr.db.BeginTxx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(groups.ErrAssignToGroup, err)
+		return errors.Wrap(auth.ErrAssignToGroup, err)
 	}
 
-	qIns := `INSERT INTO group_relations (group_id, member_id) VALUES (:group_id, :member_id)`
-	_, err = gr.db.NamedQueryContext(ctx, qIns, dbr)
-	if err != nil {
-		pqErr, ok := err.(*pq.Error)
-		if ok {
-			switch pqErr.Code.Name() {
-			case errInvalid, errTruncation:
-				return errors.Wrap(groups.ErrMalformedEntity, err)
-			case errDuplicate:
-				return errors.Wrap(groups.ErrGroupConflict, err)
-			case errFK:
-				return errors.Wrap(groups.ErrNotFound, err)
+	created := time.Now()
+
+	qIns := `INSERT INTO group_relations (group_id, member_id, type, created_at, updated_at) 
+			 VALUES( :group_id, :member_id, :type, :created_at, :updated_at)`
+
+	for _, id := range ids {
+		dbgr, err := gr.toDBGroupRelation(id, groupID, groupType)
+		if err != nil {
+			return errors.Wrap(auth.ErrAssignToGroup, err)
+		}
+		dbgr.CreatedAt = created
+		dbgr.UpdatedAt = created
```

Comment:
Same remarque as before. IMHO, this should be handled in service. Not in db related code.

Reply:
well, not sure how to handle this, I dont have object on service layer